interfaces/IMultiFeeDistributionFactory.sol

```diff
@@ -0,0 +1,17 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.0;
+
+interface IMultiFeeDistributionFactory {
+
+  // events
+  event StakerCreated(address indexed ichiVault);
```

Comment:
let's add sender to the event attributes

Reply:
Some style guidelines we try to respect universally:

1. The event log should make it theoretically possible to reconstruct the state history of the contract from logs. In practical terms, this means minimally all inputs to all mutating functions. Generally, we'll try to normalize the order of first few fields that appear all over the place. 
2. The results, a.k.a. return values.
3. Any expensive computations should be logged (and returned), a.k.a. "show your work". Internally, to negate the need to repeat the work in a higher-order function. Externally in logs so UI and caching strategies (as well as transaction debugging efforts) don't have to replicate what the contract actually did. Logs are generally cheap but if this third ideal is too expensive it can be relaxed in cases where everyone is convinced the heavy payload is useless and always will be.

Reply:
I generally agree with 1. and 2. and in my mind 2. happens to coincide with 3. i.e. the return values are typically the expensive computations e.g. return value of swap in UniswapV3 is also emitted in the Swap event.

However I don't think it's necessary to emit the farming contract address as it's deterministically computable using the factory and vault address since it was deployed with `CREATE2`.

Reply:
It's not strictly necessary but it would feel incomplete to me without it.

Reply:
hmm, I lean more towards how Uniswap V3 structured the `PoolCreated` event